### PR TITLE
HUE-8843[editor] Display metadata's describe info in left-hand column

### PR DESCRIPTION
**Pull request by hombin**
When using SparkSQL Notebook, we can't see the metadata info in left-hand column. Because the SparkSQL in NoteBook is working, I am trying to use hue's query api to fix the problem.

### CODE REVIEW

File: desktop/libs/notebook/src/notebook/connectors/hiveserver2.py [hunk 1]
```diff
@@ -435,6 +435,109 @@ def get_jobs(self, notebook, snippet, logs):
...
+          'name': table_list[1]
+        }
+        table_info_list.append(table_info)
+      spk_response = {
+        'tables_meta': table_info_list,
+      }
+      return spk_response
+
+    elif column is None:
+      _statement = u'desc `{database}`.`{table}`'.format(database=database, table=table),
```
Comment (TimKingNF) [inline desktop/libs/notebook/src/notebook/connectors/hiveserver2.py]: the _statement is tuple , not str.

Comment (hombin) [reply desktop/libs/notebook/src/notebook/connectors/hiveserver2.py]: > the _statement is tuple , not str.

In this func, _statement is the sql query string~

Comment (TimKingNF) [reply desktop/libs/notebook/src/notebook/connectors/hiveserver2.py]: Emmmm. In last character, you add a comma.  so it make mistake.

Comment (TimKingNF) [reply desktop/libs/notebook/src/notebook/connectors/hiveserver2.py]: > > the _statement is tuple , not str.
> 
> In this func, _statement is the sql query string~

Emmmm. In last character, you add a comma. so it make mistake.

Comment (hombin) [reply desktop/libs/notebook/src/notebook/connectors/hiveserver2.py]: > > > the _statement is tuple , not str.
> > 
> > 
> > In this func, _statement is the sql query string~
> 
> Emmmm. In last character, you add a comma. so it make mistake.

Oh ~ It's my mistake. Thanks for your review.

File: desktop/libs/notebook/src/notebook/connectors/hiveserver2.py [hunk 2]
```diff
@@ -434,6 +434,109 @@ def get_jobs(self, notebook, snippet, logs):
 
     return jobs
 
+  def _get_fetch_result_spk(self, _statement, snippet):
```
Comment (romainr) [inline desktop/libs/notebook/src/notebook/connectors/hiveserver2.py]: Could we move those below utilities functions (below line 730+ at least, where non API functions with _ starts)?

File: desktop/libs/notebook/src/notebook/connectors/hiveserver2.py [hunk 3]
```diff
@@ -434,6 +434,109 @@ def get_jobs(self, notebook, snippet, logs):
...
+    }
+    spk_handle.update(statement)
+    spk_snippet = {
+      "type": "sparksql",
+      'result': {
+        'handle': spk_handle,
+      }
+    }
+
+    import time
```
Comment (romainr) [inline desktop/libs/notebook/src/notebook/connectors/hiveserver2.py]: Could we move it to the top of the file?

File: desktop/libs/notebook/src/notebook/connectors/hiveserver2.py [hunk 4]
```diff
@@ -434,6 +434,109 @@ def get_jobs(self, notebook, snippet, logs):
...
+        check_query_status = check_res.get('status')
+        if check_query_status == 'available':
+          break
+        else:
+          retry_times += 1
+          if retry_times >= 30:
+            raise QueryError
+          time.sleep(1)
+      except Exception as why:
+        raise why
```
Comment (romainr) [inline desktop/libs/notebook/src/notebook/connectors/hiveserver2.py]: This is a no-op?

File: desktop/libs/notebook/src/notebook/connectors/hiveserver2.py (hunk 2 above)
Comment (romainr) [reply desktop/libs/notebook/src/notebook/connectors/hiveserver2.py]: Actually, could we move this to the beeswax.api modules?
(we hide HS2 implementation details like _autocomplete there)

Comment (hombin) [reply desktop/libs/notebook/src/notebook/connectors/hiveserver2.py]: Hello ~ romainr ~

Comment (hombin) [reply desktop/libs/notebook/src/notebook/connectors/hiveserver2.py]: Glad to receive your message, I'm tried make a encapsulation to move this api to "beewax.api".

File: desktop/libs/notebook/src/notebook/connectors/hiveserver2.py (hunk 3 above)
Comment (hombin) [reply desktop/libs/notebook/src/notebook/connectors/hiveserver2.py]: Of course~

File: desktop/libs/notebook/src/notebook/connectors/hiveserver2.py (hunk 4 above)
Comment (hombin) [reply desktop/libs/notebook/src/notebook/connectors/hiveserver2.py]: I added a Exception to return a 500 response like what "_autocomplete" do.

**Comment by romainr** [comment]
Hello, would you like to update this PR so that we could get it in?

**Comment by hombin** [comment]
> Hello, would you like to update this PR so that we could get it in?

Maybe I don't fully understand what should I do next step...

**Comment by hombin** [comment]
This is temporary approach, it is using multiple QueryAPI, hide HS2 implementation means a lot of code to rewrite.

**Pull request closed without merging**